### ISSUE TITLE
Jump between views, with search/filter

### ISSUE THREAD

**Issue by richardolsson** [zetkin/app.zetkin.org#478]
## Description
It should be possible to very quickly switch between two views when in the single-view page, by opening a menu next to the View's name and picking another view. The purpose of this is so that the user can use one view as a "source" for reading data, but save people into another view, e.g. a shortlist.

## Relevant job stories
* When I'm about to create a new view, I want to be able to use an existing one as a template or copy an existing one, so that I don't have to start from scratch
* When I'm brainstorming people to add to my view as a shortlist, I want to be able to define a Smart Search query and see the resulting data set in my view, so that I can add the ones I want to the view from there
* When I'm using a Smart Search query to populate my view, I want to be able to save it in the view so that every time I come back the view contains the people currently matching that query
* When I'm creating a shortlist of people, I want to be able to switch between different data sets in my view so that I can compare and find the ones to add

## Prerequisites
Don't start on this task until the following has been completed:

* #486 

## Requirements
- [x] A pop-over menu should be available from a small button next to the title of the view
- [x] The menu should list all other views
- [x] The menu should have a search bar to filter the list of views
- [x] Keyboard interactivity
  - [x] Arrow keys to move up/down list in menu
  - [x] Enter to navigate to selected view
  - [ ] #508

## Possible implementations
MUI [`<Popover>`](https://mui.com/components/popover/) component to select. Use existing `getViews()` to load the views (see how it's done using react-query in the view list page).

## Design specifications
![image](https://user-images.githubusercontent.com/550212/139535730-2ee68c20-b32f-462a-a93e-97cb4bf6d57b.png)

## Open questions
What should the nice-to-have keyboard shortcut be? To be discussed if implemented.

**Comment by richardolsson** [comment]
I decided not to implement a keyboard shortcut to open the menu as part of #506, because I think there is some more work needed before we're ready to start implementing keyboard shortcuts like that. See #508
